src/components/Store/Store.component.jsx

```diff
@@ -0,0 +1,82 @@
+import React, { useReducer, createContext, useContext } from 'react';
+
+import { storage } from '../../utils/storage';
+import { items } from '../../assets/mock/videolist.mock.json';
+
+import { setFavoriteList } from '../../utils/fns';
+
+const initialState = {
+  isLoading: false,
+  videos: [],
+  isDarkmodeOn: storage.get('isDarkModeOn') ?? true,
+  current: '',
+  showLogInModal: false,
+};
+
+const reducer = (
+  { isLoading, videos, isDarkmodeOn, current, showLogInModal },
+  { type, payload }
+) => {
+  switch (type) {
+    case 'toggleIsLoading':
+      return {
+        isLoading: payload || !isLoading,
+        videos,
+        isDarkmodeOn,
+        current,
+        showLogInModal,
+      };
+    case 'toggleIsDarkmodeOn':
+      storage.set('isDarkModeOn', !isDarkmodeOn);
+      return { isDarkmodeOn: !isDarkmodeOn, videos, isLoading, current, showLogInModal };
+    case 'setVideos':
+      return { videos: payload, isLoading, isDarkmodeOn, current, showLogInModal };
+    case 'setCurrent':
+      return { current: payload, isLoading, isDarkmodeOn, videos, showLogInModal };
+    case 'toggleShowLogInModal':
+      return {
+        showLogInModal: payload || !showLogInModal,
+        isLoading,
+        isDarkmodeOn,
+        videos,
+        current,
+      };
+    default:
+      throw new Error();
```

Comment:
Here you can add a message error, something like:
```suggestion
      throw new Error(`Unkown action type '${type}'`);
```
it can help you to find a bug faster